CorsixTH/Lua/utility.lua

```diff
@@ -311,8 +311,8 @@ function array_join(array, separator)
   return result
 end
 
-local function serialize_string(val)
-  local level = 0
+local function serialize_string(val, options)
+  local level = options and options.start_at_one and 1 or 0
   while string.find(val, ']' .. string.rep('=', level) .. ']') do
     level = level + 1
   end
```

Comment:
Not your fault, bu repeatedly walking over a potentially long string is  a bad idea. The better solution is to catch all level in one search by using a regular expression.